limits.c

```diff
@@ -111,92 +110,96 @@ void limits_go_home(uint8_t cycle_mask)
 {
   if (sys.abort || !cycle_mask) { return; } // Block if system reset has been issued.
 
+  float target[N_AXIS];
+  float homing_rate;
+  // Determine travel distance to the furthest homing switch based on user max travel settings.
+  float min_seek_rate = 1e9; //arbitrary maximum=1km/s, will be reduced by axis setting below
+  float max_travel = 0;
   // Initialize homing in search mode to quickly engage the specified cycle_mask limit switches.
   uint8_t approach = ~0;  //approach has all bits set (negative dir) or none (positive)
-  float homing_rate;
   uint8_t idx;
-  uint8_t n_cycle = (2*N_HOMING_LOCATE_CYCLE+1);
-  float target[N_AXIS];
+  uint8_t n_cycle = (2*N_HOMING_LOCATE_CYCLE);
+  uint8_t n_active_axis = 0;
 
   uint8_t flipped = settings.homing_dir_mask>>X_DIRECTION_BIT;  //assumes keyme configuration.
-  //replace with an instance of `if (bitistrue(h_d_m,X_DIRECTION_BIT)) { flipped|=1<<X_AXIS;}` 
-  //for each axis if the bits line up differently
-
-  // Determine travel distance to the furthest homing switch based on user max travel settings.
-  float min_seek_rate = 1e9; //arbitrary maximum=1km/s, will be reduced by axis setting below
-  float max_travel = 0;
+  uint8_t axislock = 0; 
 
   for (idx=0; idx<N_AXIS; idx++){
     if (bit_istrue(cycle_mask,bit(idx))) {
+      n_active_axis++;
+      axislock |= (1<<(X_STEP_BIT+idx)); //assumes axes are in bit order.
       max_travel = max(max_travel,settings.max_travel[idx]);
       min_seek_rate = min(min_seek_rate,settings.homing_seek_rate[idx]);
     }
   }
   max_travel *= HOMING_AXIS_SEARCH_SCALAR; // Ensure homing switches engaged by over-estimating max travel.
-  max_travel += settings.homing_pulloff;
-  homing_rate = min_seek_rate;
+  homing_rate = min_seek_rate * sqrt(n_active_axis); //Adjust so individual axes all move at homing rate.
 
   plan_reset(); // Reset planner buffer to zero planner current position and to clear previous motions.
 
   do {
     // Set target location and rate for active axes.
     // and reset homing axis locks based on cycle mask.  
-    uint8_t axislock = 0; 
-    uint8_t n_active_axis = 0;
+    float travel = (approach)?max_travel:settings.homing_pulloff; //limit travel away from the switch to the pulloff distance.
```

Comment:
yes.  "homing_pulloff" is the distance between the limit switch and the point we call 0. 
If it is approaching the limit, it can go up to max_travel + homing_pulloff.  It if is not approaching, it can only go twice the pulloff.